src/olympia/promoted/models.py

```diff
@@ -96,26 +96,11 @@ def approve_for_version(self, version):
     def approve_for_addon(self):
         """This sets up the addon as approved for the current promoted group.
 
-        The current version will be signed for approval, and if there's special
-        signing needed for that group the version will be resigned."""
-        from olympia.lib.crypto.tasks import sign_addons
+        The current version will be signed for approval."""
 
         if not self.addon.current_version:
             return
         self.approve_for_version(self.addon.current_version)
-        if self.group.autograph_signing_states:
-            sign_addons([self.addon.id], send_emails=False)
```

Comment:
This was - thankfully - dead code. The only place outside tests that calls `approve_for_addon()` is in `save()` (a few lines below if you expand the diff). It only calls it if the promoted group has `immediate_approval`, and in here we checked if the group has some `autograph_signing_states`.

There is no promoted group that has both of these set. We have groups for which the promotion is immediate (`Spotlight`, because add-ons are added to that group after review anyway) and groups that cause the signature to be special (`Recommended`/`Line` and the obsolete `Verified`/`Sponsored`) but we never combine the two - we decided long ago that to "gain" a promotion with a special signature you need a new version to be explicitly uploaded, we never want to resign existing ones.